lib/hanami/action/mime.rb

```diff
@@ -207,7 +212,14 @@ def format
       #     end
       #   end
       def content_type
-        @content_type || default_response_type || accepts || default_content_type || DEFAULT_CONTENT_TYPE
+        return @content_type if @content_type
+
+        if accept_header?
+          type = get_mime_type_from_accept_header
```

Comment:
Also (not related to this PR), is there a size limit on the `Accept` header at some level, or can we simply make the CPU spin by supplying a very long list of random values?